local-cli/runAndroid/runAndroid.js

```diff
@@ -123,15 +123,28 @@ function buildAndRun(args, reject) {
 }
 
 function startServerInNewWindow() {
+  var yargV = require('yargs').argv;
+
   const launchPackagerScript = path.resolve(
     __dirname, '..', '..', 'packager', 'launchPackager.command'
   );
 
   if (process.platform === 'darwin') {
-    child_process.spawnSync('open', [launchPackagerScript]);
+    if (!yargV.open) {
```

Comment:
returns to avoid more nested `if else` when nothing is being done afterwards any how.